mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
I tend to agree that this way feels strange. This was the suggested solution in the JIRA ticket, but I'd be happy to update `Vectors.sparse` instead. I considered it, but wasn't sure it made sense to strip out explicit zeros unless it was actually needed.